tests/pytest/test_parameters.py

```diff
@@ -326,6 +326,114 @@ def test_project_separation(self):
+        env1_list = proj_cmd + param_ls
+        env2_list = proj_cmd + f"--env '{env_name2}'" + param_ls
+        env3_list = proj_cmd + f"--env '{env_name3}'" + param_ls
+
+        # see that values are inherited in the different environments
+        result = self.run_cli(cmd_env, env1_list)
+        self.assertIn(f"{var1_name},{var1_value1},{env_name1}", result.out())
+        self.assertIn(f"{var2_name},{var2_value1},{env_name1}", result.out())
+
+        result = self.run_cli(cmd_env, env2_list)
+        self.assertIn(f"{var1_name},{var1_value1},{env_name1}", result.out())
+        self.assertIn(f"{var2_name},{var2_value1},{env_name1}", result.out())
+
+        result = self.run_cli(cmd_env, env3_list)
+        self.assertIn(f"{var1_name},{var1_value1},{env_name1}", result.out())
+        self.assertIn(f"{var2_name},{var2_value1},{env_name1}", result.out())
+
+        # add the parameters for the second environment
+        self.set_param(cmd_env, proj_name, var1_name, var1_value2, env=env_name2)
+        self.set_param(cmd_env, proj_name, var2_name, var2_value2, True, env=env_name2)
+
+        # see that values are inherited in the different environments
+        result = self.run_cli(cmd_env, env1_list)
+        self.assertIn(f"{var1_name},{var1_value1},{env_name1}", result.out())
+        self.assertIn(f"{var2_name},{var2_value1},{env_name1}", result.out())
+
+        result = self.run_cli(cmd_env, env2_list)
+        self.assertIn(f"{var1_name},{var1_value2},{env_name2}", result.out())
+        self.assertIn(f"{var2_name},{var2_value2},{env_name2}", result.out())
+
+        result = self.run_cli(cmd_env, env3_list)
+        self.assertIn(f"{var1_name},{var1_value2},{env_name2}", result.out())
+        self.assertIn(f"{var2_name},{var2_value2},{env_name2}", result.out())
+
+        # add the parameters for the third environment
+        self.set_param(cmd_env, proj_name, var1_name, var1_value3, env=env_name3)
+        self.set_param(cmd_env, proj_name, var2_name, var2_value3, True, env=env_name3)
+
+        # see that values are inherited in the different environments
+        result = self.run_cli(cmd_env, env1_list)
+        self.assertIn(f"{var1_name},{var1_value1},{env_name1}", result.out())
+        self.assertIn(f"{var2_name},{var2_value1},{env_name1}", result.out())
+
+        result = self.run_cli(cmd_env, env2_list)
+        self.assertIn(f"{var1_name},{var1_value2},{env_name2}", result.out())
+        self.assertIn(f"{var2_name},{var2_value2},{env_name2}", result.out())
+
+        result = self.run_cli(cmd_env, env3_list)
+        self.assertIn(f"{var1_name},{var1_value3},{env_name3}", result.out())
+        self.assertIn(f"{var2_name},{var2_value3},{env_name3}", result.out())
+
+        docker_cmd = " param export docker -s"
+        result = self.run_cli(cmd_env, proj_cmd + docker_cmd)
+        self.assertEqual(result.return_value, 0)
+        self.assertEqual(result.out(), f"""\
+{var1_name.upper()}={var1_value1}
```

Comment:
Order does not matter so one could test to make sure each line is present.

Reply:
Let's chat about this... We do order the `param export` output on the server side, although the order will likely change soon. We should understand what cli tables/lists we can depend on the order of and probably fix all of them.